File: app/src/main/java/codes/wokstym/cookingrecipes/models/ShoppingListDto.java
package codes.wokstym.cookingrecipes.models;

import java.io.Serializable;
import java.util.List;

import androidx.annotation.NonNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ShoppingListDto implements Serializable {
    private List<IngredientDto> ingredientList;

    @NonNull
    @Override
    public String toString() {
        StringBuilder result = new StringBuilder();

        for (IngredientDto ingredientQuantity : ingredientList) {
            result.append(ingredientQuantity.getName())
                    .append(" - ")
                    .append(ingredientQuantity.getAmount())
                    .append(" ")
                    .append(ingredientQuantity.getUnit().getName())
                    .append(" \n");
        }

        return result.toString();
    }
}

File: app/src/main/java/codes/wokstym/cookingrecipes/models/ShoppingListDto.kt
package codes.wokstym.cookingrecipes.models

import android.content.Context
import java.io.Serializable

class ShoppingListDto(val ingredientList: List<IngredientDto>) : Serializable {

    fun formatClipboard(context: Context): String =
            ingredientList.joinToString(separator = "\n") {
                "${it.name} - ${it.amount} ${context.getString(it.unit.nameResourceId)}"
            }

}
